src/DIRAC/Resources/IdProvider/OAuth2IdProvider.py

```diff
@@ -222,30 +222,34 @@ def revokeToken(self, token=None, token_type_hint="refresh_token"):
             return S_ERROR(repr(e))
         return S_OK()
 
-    def exchangeGroup(self, group):
+    def exchangeToken(self, group=None, scope=None):
```

Comment:
This module has now methods `exchangeToken` and `exchange_token`, which I find confusing and possibly misleading. Can you give a better name?

And BTW, at line 217 (`tokn = self.token.get(token_type_hint)`) I guess there's a typo.

Reply:
thanks, bug fixed.
As for the names of the methods, I think I need to do a little refactoring. in progress

Reply:
I put the auxiliary methods (`exchange_token`, `_exchange_token`) in a separate class to emphasize that this part is not yet implemented in the original library. So only the DIRAC-adapted method remains `exchangeToken`.